openhtf/io/output/console_summary.py

```diff
@@ -0,0 +1,113 @@
+"""Module to display test summary on console."""
+import os
+import sys
+
+from openhtf import util
+from openhtf.io import test_record
+from openhtf.util import measurements as measurement
+
+
+class ConsoleSummary():
+  """Print test results with failure info on console. """
+
+  # pylint: disable=invalid-name
+  def __init__(self):
+    if os.name == 'posix':    #Linux and Mac
+      self.RED = '\033[91m'
+      self.GREEN = '\033[92m'
+      self.ORANGE = '\033[93m'
+      self.RESET = '\033[0m'
+      self.BOLD = '\033[1m'
+    else:
+      self.RED = ""
+      self.GREEN = ""
+      self.ORANGE = ""
+      self.RESET = ""
+      self.BOLD = ""
+
+    self.color_table = {
+        test_record.Outcome.PASS:self.GREEN,
+        test_record.Outcome.FAIL:self.RED,
+        test_record.Outcome.ERROR:self.ORANGE,
+        test_record.Outcome.TIMEOUT:self.ORANGE,
+    }
+  # pylint: enable=invalid-name
+
+  def __call__(self, record):
+    indent2 = ''.join(('\n', ' '*2))
+    indent4 = ''.join(('\n', ' '*4))
+
+    # get status string from outcome enum
+    status = str(record.outcome)[-5:]
+    if status[0] == '.':
+      status = status[1:]
+
+    color = self.color_table[record.outcome]
+
+    output_lines = []
+    msg = ''.join((color,
+                   self.BOLD,
+                   os.path.basename(sys.argv[0]),
```

Comment:
It's not clear what `os.path.basename(sys.argv[0])` represents in this context. Looks like it's probably the path to the python interpreter being used?